### PR TITLE
Added support to change the current time zone(#1y8m51y)

### PR DESCRIPTION
**Pull request by disha1202**
Screenshot from 2022-01-11 17-29-20

### CODE REVIEW

File: src/views/Settings.vue [hunk 1]
```diff
@@ -54,6 +59,9 @@ export default defineComponent({
       currentFacility: 'user/getCurrentFacility',
     })
   },
+  mounted(){
+    console.log(this.userProfile);
```
Comment (adityasharma7) [inline src/views/Settings.vue:63]: It seems you missed a console log

File: src/views/TimezoneModal.vue [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    </ion-toolbar>
+    <ion-toolbar>
+      <ion-searchbar @ionFocus="selectSearchBarText($event)" :placeholder="$t('Search time zones')"  v-model="queryString" @ionInput="findTimeZone()"></ion-searchbar>
+    </ion-toolbar>
+  </ion-header>
+
+  <ion-content class="ion-padding">
+    <!-- Empty state -->
+    <div class="empty-state" v-if="filteredTimeZones.length === 0">
+      <p>{{ $t("No time zone found")}}</p>
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:19]: Add internationalized text to en.json

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Comment by adityasharma7** [comment]
@disha1202 
Handle case for timezone mismatch
https://github.com/hotwax/preorder/blob/main/src/App.vue#L40

File: src/views/Settings.vue (hunk 1 above)
Comment (disha1202) [reply src/views/Settings.vue]: Remove it.

File: src/views/TimezoneModal.vue (hunk 2 above)
Comment (disha1202) [reply src/views/TimezoneModal.vue:19]: Added internationalized text to en.json

**Comment by disha1202** [comment]
> @disha1202 Handle case for timezone mismatch https://github.com/hotwax/preorder/blob/main/src/App.vue#L40

Handled case for mismatched timezone.

File: src/App.vue [hunk 3]
```diff
@@ -23,6 +24,29 @@ export default defineComponent({
     }
   },
   methods: {
+        async timeZoneDifferentAlert(payload: any) {
+      const alert = await alertController.create({
```
Comment (adityasharma7) [inline src/App.vue:28]: Indentation issues

File: src/App.vue [hunk 4]
```diff
@@ -41,19 +65,21 @@ export default defineComponent({
...
   },
-  async mounted() {
-    this.loader = await loadingController
-      .create({
-        message: this.$t("Click the backdrop to dismiss."),
-        translucent: true,
-        backdropDismiss: true
-      });
+  mounted() {
+    emitter.on('timeZoneDifferent', this.timeZoneDifferentAlert);
```
Comment (adityasharma7) [inline src/App.vue:69]: This needs to be emitted from userprofile action
https://github.com/hotwax/preorder/blob/5f56d984dbcf531c116b81035ad70abecc18bea3/src/store/modules/user/actions.ts#L58

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 4 above)
Comment (disha1202) [reply src/App.vue:69]: Yes sir, we are emitting this from user getprofile action.
https://github.com/hotwax/ionic-sdk/blob/main/src/store/modules/user/actions.ts#L58

File: src/App.vue (hunk 3 above)
Comment (disha1202) [reply src/App.vue:28]: Fixed indentation issues

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**